runtime/src/main/cpp/Porting.h

```diff
@@ -94,6 +99,23 @@ do {                          \
 } while(0)
 #endif
 
+template <typename F>
+// TODO: Use invoke_result without && when compiled with C++17
```

Comment:
Can't catch this comment. For C++14 and above a simple `auto` return would be sufficient. If you mean "stay with c++11 but be ready for 14-17" you may use a form `auto + trailing return type` and say "TODO: remove trailing for c++14"
```
auto tryOrTerminate(F&& tryClause) -> typename std::result_of<F&&()>::type {
```
As for me, `auto` seems more readable, but it's only my personal preference.

Reply:
To be honest, I don't like type inference in function type signatures. I use `auto` for local variables only. But then again, this is just my personal preference.

As for what I meant: `result_of` is deprecated in c++17 and removed in c++20, having been replaced with `invoke_result`.